src/test/scala/heuristics/ImplicitClassExtensionTest.scala

```diff
@@ -0,0 +1,134 @@
+package heuristics
+
+import extractor.{ExtractImplicits, SemanticCtx}
+import framework.SemanticdbTest
+import org.langmeta.semanticdb.{ResolvedName, Synthetic}
+
+class ImplicitClassExtensionTest extends SemanticdbTest {
+
+  /**
+    * Define a method to identify the definition of the
+    * Implicit class extension pattern.
+    *
+    * This can be done after extraction in an R script since, as it is shown,
+    * we already have the necessary data.
+    * @param ctx
+    */
+  def checkDefinition(ctx: SemanticCtx): Unit = {
+    val res = ExtractImplicits(ctx)
+    res.implicits should have size 1
+    val classDeclaration = res.implicits.head
+    classDeclaration.kind shouldEqual "class"
+  }
+
+  /**
+    * Define some conditions that all usages of this pattern must use.
+    * These should be sufficient conditions to determine that a synthetic wrapper
+    * corresponds to this pattern's usage.
+    *
+    * Note that there is some string parsing involved, which I think should be done
+    * after the data is in the database.
+    *
+    * Note also that we need to capture a sort of Synthetic that we didn't before,
+    * which will likely involve an extension of the model and the db schema.
+    *
+    * @param ctx
+    */
+  def checkUsage(ctx: SemanticCtx): Unit = {
+    // See that the inserted call matches the expected fqn.of.constructor(*) pattern
+    val usages = ctx.index.synthetics.filter(_.text.matches("""(\.?[\[\w\]]*)+\(\*\)"""))
+    usages should not be empty
+    usages.foreach { usage =>
```

Comment:
Isn't the same thing above?

Reply:
I;m not really getting this, but cannot it be simplified a bit with just plain matching of case classes?

Reply:
It probably can, yes, but I was trying to imagine what a possible R script would look like that classified all the data.

In this particular instance, it is trying to identify the usage of an implicit class conversion, as described in the last paragraph of this section of the wiki: https://github.com/PRL-PRG/scalafix-rule-workshop/wiki/Patterns:-Heuristics-and-intuitions#application